compiler/modules/types/core.bal

```diff
@@ -653,10 +649,16 @@ public function intersect(SemType t1, SemType t2) returns SemType {
     return createComplexSemType(all, subtypes);    
 }
 
-public function intersectMemberSemTypes(Env env, MemberSemType t1, MemberSemType t2) returns MemberSemType {
-    // JBUG #37994 cannot use mapping binding pattern with readonly records
-    CellAtomicType cat = intersectCellAtomicType(<CellAtomicType>cellAtomicType(t1), <CellAtomicType>cellAtomicType(t2));
-    return <MemberSemType>cellContaining(env, cat.ty, cat.mut);
+public function intersectMemberTypes(Env env, SemType t1, SemType t2) returns SemType {
```

Comment:
Here I have made the arguments `SemType` instead of `MemberSemType` as we support the intersection for both normal SemTypes and MemberSemTypes in this function. With that the function name `intersectMemberSemTypes` was misleading. Therefore, changed that to `intersectMemberTypes`.